--- REL_17_5_WASM/src/bin/pg_dump/pg_dump.c
+++ REL_17_5_WASM-pglite/src/bin/pg_dump/pg_dump.c
@@ -30,13 +30,6 @@
  *-------------------------------------------------------------------------
  */
 #include "postgres_fe.h"
-#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
-#ifdef quote_all_identifiers
-#undef quote_all_identifiers
-#endif
-#define fe_utils_quote_all_identifiers quote_all_identifiers
-static bool quote_all_identifiers;
-#endif
 
 #include <unistd.h>
 #include <ctype.h>
@@ -76,6 +69,30 @@
 #include "pg_dump.h"
 #include "storage/block.h"
 
+#if !defined(__wasi__) && !defined(__EMSCRIPTEN__)
+extern PGDLLIMPORT bool fe_utils_quote_all_identifiers;
+#else
+#if defined(WASM)
+#ifdef errno
+    #undef errno
+    int errno;
+#endif
+
+pid_t fork(void) {
+    return -1;
+}
+
+#   include <wasi/api.h>
+
+__attribute__((export_name("sdk_fd_seek")))
+int sdk_fd_seek(int fd, int  offset,int  whence, unsigned long long *retptr) {
+    puts("sdk_fd_seek called !");
+    return __wasi_fd_seek(fd, offset, whence, retptr);
+}
+
+#endif
+#endif
+
 typedef struct
 {
 	Oid			roleoid;		/* role's OID */
@@ -432,7 +449,7 @@
 		{"lock-wait-timeout", required_argument, NULL, 2},
 		{"no-table-access-method", no_argument, &dopt.outputNoTableAm, 1},
 		{"no-tablespaces", no_argument, &dopt.outputNoTablespaces, 1},
-		{"quote-all-identifiers", no_argument, (int *)(&fe_utils_quote_all_identifiers), true},
+		{"quote-all-identifiers", no_argument, (int *)(&fe_utils_quote_all_identifiers), 1},
 		{"load-via-partition-root", no_argument, &dopt.load_via_partition_root, 1},
 		{"role", required_argument, NULL, 3},
 		{"section", required_argument, NULL, 5},
@@ -459,9 +476,7 @@
 
 		{NULL, 0, NULL, 0}
 	};
-#if defined(__wasi__)
-chdir("/");
-#endif
+
 	pg_logging_init(argv[0]);
 	pg_logging_set_level(PG_LOG_WARNING);
 	set_pglocale_pgservice(argv[0], PG_TEXTDOMAIN("pg_dump"));
@@ -838,12 +853,13 @@
 	fout->maxRemoteVersion = (PG_VERSION_NUM / 100) * 100 + 99;
 
 	fout->numWorkers = numWorkers;
-
+puts("# 832:" __FILE__ ": ConnectDatabase\r\n");
 	/*
 	 * Open the database using the Archiver, so it knows about it. Errors mean
 	 * death.
 	 */
 	ConnectDatabase(fout, &dopt.cparams, false);
+puts("# 838:" __FILE__ ": ConnectDatabase->setup_connection\r\n");
 	setup_connection(fout, dumpencoding, dumpsnapshot, use_role);
 
 	/*
