--- REL_17_5_WASM/src/bin/pg_dump/pg_dump.c
+++ pglite-REL_17_5_WASM/src/bin/pg_dump/pg_dump.c
@@ -843,12 +843,12 @@
 	 * Open the database using the Archiver, so it knows about it. Errors mean
 	 * death.
 	 */
-puts("# 813 : " __FILE__);
+puts("# 813 : " __FILE__ "\r\n");
     //setup();
 	ConnectDatabase(fout, &dopt.cparams, false);
-puts("# 815 : " __FILE__);
+puts("# 815 : " __FILE__ "\r\n");
 	setup_connection(fout, dumpencoding, dumpsnapshot, use_role);
-puts("# 817 : " __FILE__);
+puts("# 817 : " __FILE__ "\r\n");
 	/*
 	 * On hot standbys, never try to dump unlogged table data, since it will
 	 * just throw an error.
@@ -1295,7 +1295,7 @@
 	 * Quote all identifiers, if requested.
 	 */
 	if (fe_utils_quote_all_identifiers)
-		ExecuteSqlStatement(AH, "SET fe_utils_quote_all_identifiers = true");
+		ExecuteSqlStatement(AH, "SET quote_all_identifiers = true");
 
 	/*
 	 * Adjust row-security mode, if supported.
