--- REL_17_5_WASM/src/backend/utils/error/elog.c
+++ REL_17_5_WASM-pglite/src/backend/utils/error/elog.c
@@ -348,12 +348,16 @@
 	ErrorData  *edata;
 	bool		output_to_server;
 	bool		output_to_client = false;
+#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
	int			i;
 
 	/*
 	 * Check some cases in which we want to promote an error into a more
 	 * severe error.  None of this logic applies for non-error messages.
 	 */
+
 	if (elevel >= ERROR)
 	{
 		/*
@@ -394,7 +398,7 @@
 		for (i = 0; i <= errordata_stack_depth; i++)
 			elevel = Max(elevel, errordata[i].elevel);
 	}
-
+#endif
 	/*
 	 * Now decide whether we need to process this report at all; if it's
 	 * warning or less and not enabled for logging, just return false without
@@ -539,7 +543,7 @@
 		 */
 
 		recursion_depth--;
 		PG_RE_THROW();
 	}
 
 	/* Emit the message to the right places */
@@ -587,7 +596,11 @@
 		 * FATAL termination.  The postmaster may or may not consider this
 		 * worthy of panic, depending on which subprocess returns it.
 		 */
+#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
		proc_exit(1);
+#endif
 	}
 
 	if (elevel >= PANIC)
@@ -697,6 +710,7 @@
 	 */
 	if (edata->elevel >= ERROR)
 	{
 		errfinish(filename, lineno, funcname);
 		pg_unreachable();
 	}
