--- REL_17_5_WASM/src/backend/utils/error/elog.c
+++ pglite-REL_17_5_WASM/src/backend/utils/error/elog.c
@@ -543,13 +543,8 @@
 		 */
 
 		recursion_depth--;
-#if 0 //defined(__EMSCRIPTEN__) || defined(__wasi__)
-        fprintf(stderr, "# 547: PG_RE_THROW(ERROR : %d) ignored\n", recursion_depth);
-        trap();
-#else
         fprintf(stderr, "# 549: PG_RE_THROW(ERROR : %d)\n", recursion_depth);
 		PG_RE_THROW();
-#endif
 	}
 
 	/* Emit the message to the right places */
