--- REL_17_4_WASM/src/backend/tcop/postgres.c
+++ pglite-REL_17_4_WASM/src/backend/tcop/postgres.c
@@ -110,6 +110,7 @@
 #if (defined(__EMSCRIPTEN__) || defined(__wasi__))
 #if !defined(PGL_MAIN)
     volatile int cma_rsize = 0;
+    volatile bool sockfiles = false;
 #endif // PGL_MAIN
 bool quote_all_identifiers = false;
 FILE* SOCKET_FILE = NULL;
@@ -5146,7 +5147,7 @@
 					 (long) sys.tv_sec,
 					 (long) sys.tv_usec);
 #ifndef WIN32
-
+#if !defined(__wasi__)
 	/*
 	 * The following rusage fields are not defined by POSIX, but they're
 	 * present on all current Unix-like systems so we use them without any
@@ -5188,6 +5189,7 @@
 					 r.ru_nvcsw - Save_r.ru_nvcsw,
 					 r.ru_nivcsw - Save_r.ru_nivcsw,
 					 r.ru_nvcsw, r.ru_nivcsw);
+#endif  /* !__wasi__ */
 #endif							/* !WIN32 */
 
 	/* remove trailing newline */
